lib/racecar/runner.rb

```diff
@@ -95,7 +95,7 @@ def run
       end
     ensure
       producer.close
-      Racecar::Datadog.close if Object.const_defined?("Racecar::Datadog")
+      Racecar::Datadog.close if config.datadog_enabled
```

Comment:
This was causing some errors when running the integration tests locally. I think this is because `Racecar::Datadog` is defined as a result of previously run unit tests and `#close` is called which results in this error.
```
9) kubernetes probes liveness probe when the probe is disabled does not touch the file
   Failure/Error: Racecar::Datadog.statsd.send(type, *args, tags: tags)

   ArgumentError:
     Start sender first
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/dogstatsd-ruby-5.5.0/lib/datadog/statsd/sender.rb:64:in `add'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/dogstatsd-ruby-5.5.0/lib/datadog/statsd/forwarder.rb:75:in `send_message'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/dogstatsd-ruby-5.5.0/lib/datadog/statsd.rb:430:in `send_stats'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/dogstatsd-ruby-5.5.0/lib/datadog/statsd.rb:223:in `histogram'
   # ./lib/racecar/datadog.rb:90:in `emit'
   # ./lib/racecar/datadog.rb:83:in `block (2 levels) in <class:StatsdSubscriber>'
   # ./lib/racecar/datadog.rb:179:in `main_loop'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/activesupport-6.1.7.3/lib/active_support/subscriber.rb:150:in `finish'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/activesupport-6.1.7.3/lib/active_support/notifications/fanout.rb:157:in `finish'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/activesupport-6.1.7.3/lib/active_support/notifications/fanout.rb:63:in `block in finish'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/activesupport-6.1.7.3/lib/active_support/notifications/fanout.rb:63:in `each'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/activesupport-6.1.7.3/lib/active_support/notifications/fanout.rb:63:in `finish'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/activesupport-6.1.7.3/lib/active_support/notifications/instrumenter.rb:45:in `finish_with_state'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/activesupport-6.1.7.3/lib/active_support/notifications/instrumenter.rb:30:in `ensure in instrument'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/activesupport-6.1.7.3/lib/active_support/notifications/instrumenter.rb:30:in `instrument'
   # /home/admin/.rvm/gems/ruby-2.7.2/gems/activesupport-6.1.7.3/lib/active_support/notifications.rb:203:in `instrument'
   # ./lib/racecar/instrumenter.rb:18:in `instrument'
   # ./lib/racecar/runner.rb:72:in `block in run'
   # ./lib/racecar/runner.rb:67:in `loop'
   # ./lib/racecar/runner.rb:67:in `run'
   # ./lib/racecar/cli.rb:70:in `run'
   # ./spec/support/integration_helper.rb:31:in `block in start_racecar'
   # ------------------
   # --- Caused by: ---
   # ArgumentError:
   #   Start sender first
   #   /home/admin/.rvm/gems/ruby-2.7.2/gems/dogstatsd-ruby-5.5.0/lib/datadog/statsd/sender.rb:64:in `add'
```

Using `config.datdog_enabled` as the conditional fixes this. It doesn't change the running behavior and allows us to run the full rspec suite without running into side effects from other tests.